src/python/sdk/econia_sdk/utils/decimals.py

```diff
@@ -0,0 +1,106 @@
+import math
+from decimal import Decimal
+
+
+def get_lot_size_integer(smallest_decimal_unit: str, coin_decimals: int) -> int:
+    """
+    Given a decimal representation of the smallest possible unit and the
+    whole unit's decimals (e.g. ETH has 18 decimals), return the lot size.
+
+    Parameters:
+    * `smallest_decimal_unit`: Decimal (e.g. "0.001" for one thousandth) of
+      the smallest possible unit relative to a whole unit.
+    * `coin_decimals`: The number of decimals one whole unit of the coin
+      has (e.g. USDC has 6 decimals, ETH has 18).
+    """
+    smallest_decimal = Decimal(smallest_decimal_unit)
+    smallest_subunits = 10 ** (coin_decimals + math.log10(smallest_decimal))
+    if smallest_subunits < 1:
+        raise ValueError("Decimal unit too small to represent with 1 subunit")
+    return math.ceil(smallest_subunits)
+
+
+def get_tick_size_integer(
+    smallest_decimal_unit: str, coin_decimals: int
+) -> int:
+    """
+    Given a decimal representation of the smallest possible unit and the
+    whole unit's decimals (e.g. ETH has 18 decimals), return the tick size.
+
+    Parameters:
+    * `smallest_decimal_unit`: Decimal (e.g. "0.001" for one thousandth) of
+      the smallest possible unit relative to a whole unit.
+    * `coin_decimals`: The number of decimals one whole unit of the coin
+      has (e.g. USDC has 6 decimals, ETH has 18).
+    """
+    return get_lot_size_integer(smallest_decimal_unit, coin_decimals)
+
+
+def get_min_size_integer(
+    smallest_decimal_size: str, base_coin_decimals: int, lot_size: int
+) -> int:
+    """
+    Returns the minimum size, in number of lots, to represent the smallest
+    decimal size according to the base coin's decimals and lot size.
+
+    Parameters:
+    * `smallest_decimal_size`: Decimal (e.g. "0.001" for one thousandth) of
+      the smallest possible size relative to a whole unit. Note that this
+      maybe different (larger) than the smallest possible unit (lot size).
+      This should be of the base coin, not the quote coin!
+    * `base_coin_decimals`: The number of decimals one whole unit of the
+      base has (e.g. USDC has 6 decimals, ETH has 18).
+    * `lot_size`: The size in subunits of one lot of base coin.
+    """
+    smallest_decimal = Decimal(smallest_decimal_size)
+    smallest_subunits = (
+        smallest_decimal * (10**base_coin_decimals)
```

Comment:
I'm surprised the formatter gave `(10**base_coin_decimals)`...

Reply:
I ran it myself and got the same result, so I guess black removes spaces in exponentiation like this

However there were other changes, so will you please re-run the formatter once more after addressing the above changes?